From: Ximenes Zalteca <ximenes@mythic.net>
Subject: qmail-pop3d and CAPA
Date: Sat, 3 Apr 1999 23:11:44 -0500

	I've noticed that beta versions of Eudora (which my clients 
will eventually update to when they are released, I presume) tell the 
pop server 'CAPA' to determine what commands are supported when they 
log in. I'm still undecided about whether or not this -should- be 
supported, but I went ahead and modified the existing patch (that 
just returns 'APOP') to function more like Eudora expects anyway.

	Basically, I took a look at what qpopper says (I presume that 
if anything returns the correct information, it's qpopper) and made 
qmail-popup say the same thing (minus strange stuff that I don't 
think it supports).

	You may want to remove the APOP line if you don't actually 
-use- APOP, Other than that, the rest of the commands are supported 
by vanilla qmail-popup.

Xi

--- qmail-1.03/qmail-popup.c Mon Jun 15 03:53:16 1998
+++ new-qmail-1.03/qmail-popup.c     Thu Dec 10 16:52:59 1998
@@ -155,11 +155,23 @@
   *space++ = 0;
   doanddie(arg,space - arg,space);
 }
+void pop3_capa(arg) char *arg;
+{
+  puts("+OK Capability list follows\r\n");
+  puts("TOP\r\n");
+  puts("UIDL\r\n");
+  puts("LAST\r\n");
+  puts("USER\r\n");
+  puts("APOP\r\n");
+  puts(".\r\n");
+  flush();
+}
 
 struct commands pop3commands[] = {
   { "user", pop3_user, 0 }
 , { "pass", pop3_pass, 0 }
 , { "apop", pop3_apop, 0 }
+, { "capa", pop3_capa, 0 }
 , { "quit", pop3_quit, 0 }
 , { "noop", okay, 0 }
 , { 0, err_authoriz, 0 }
